src/structured_config.py

```diff
@@ -39,6 +39,15 @@ class CompressionType(str, Enum):
     PRODUCER = "producer"
 
 
+class LogLevel(str, Enum):
+    """Enum for the `log_level` field."""
+
+    INFO = "INFO"
+    WARNING = "WARNING"
+    ERROR = "ERROR"
+    DEBUG = "DEBUG"
```

Comment:
**thought:** Personally I dislike the Enum pattern of `Class.VAR` which always just equals `'VAR': str` anyway, it seems redundant and obfuscates types.

It's non-blocking, but we might consider just having `LOG_LEVEL = Literal["INFO","WARNING","ERROR","DEBUG"]` and type hinting `LOG_LEVEL` everywhere instead.

If I'm right, I **think** that means we can get rid of the `log_level_values` validator further down, as Pydantic should infer the type? Not certain on that.

Reply:
I used the same pattern that was used for other fields (e.g. CompressionTypes, LogCleanupPolicy), so I would rather keep things consistent for the time being. 

TBH, I don't mind Enum: they avoid using strings throughout the code and centralize constants in one single place. But I suppose this would be a different, broader discussion, around common patterns we'd like to adopt